chandra_aca/plot.py

```diff
@@ -0,0 +1,286 @@
+            width=acq_star['halfw'] * 2,
+            height=acq_star['halfw'] * 2,
+            color='blue',
+            fill=False)
+        ax.add_patch(box)
+    for mon_box in mon:
+        # starcheck convention was to plot monitor boxes at 2X halfw
+        box = plt.Rectangle(
+            (mon_box['yang'] - (mon_box['halfw'] * 2),
+             mon_box['zang'] - (mon_box['halfw'] * 2)),
+            width=mon_box['halfw'] * 4,
+            height=mon_box['halfw'] * 4,
+            color='orange',
+            fill=False)
+        ax.add_patch(box)
+    ax.scatter(fid['yang'], fid['zang'],
+               facecolors='none',
+               edgecolors='red',
+               linewidth=.5,
+               marker='o',
+               s=175)
+    ax.scatter(fid['yang'], fid['zang'],
+               facecolors='none',
+               edgecolors='red',
+               marker='+',
+               linewidth=.5,
+               s=175)
+
+
+def _plot_field_stars(ax, stars, attitude, red_mag_lim=None, bad_stars=None):
+    """
+    Plot plot field stars in yang and zang on the supplied
+    axes object in place.
+
+    :param ax: matplotlib axes
+    :param stars: astropy.table compatible set of records of agasc entries of stars
+    :param attitude: Quaternion-compatible attitude
+    :param red_mag_lim: faint limit
+    :param bad_stars: boolean mask of stars to be plotted in red
+    """
+    stars = Table(stars)
+    quat = Quaternion.Quat(attitude)
+
+    if bad_stars is None:
+        bad_stars = np.zeros(len(stars), dtype=bool)
+
+    if 'yang' not in stars.colnames or 'zang' not in stars.colnames:
+        # Add star Y angle and Z angle in arcsec to the stars table.
+        # radec2yagzag returns degrees.
+        yags, zags = radec2yagzag(stars['RA_PMCORR'], stars['DEC_PMCORR'], quat)
+        stars['yang'] = yags * 3600
+        stars['zang'] = zags * 3600
+
+    # Initialize array of colors for the stars, default is black
+    colors = np.zeros(len(stars), dtype='S20')
+    colors[:] = 'black'
+
+    colors[bad_stars] = BAD_STAR_COLOR
+
+    if red_mag_lim:
+        # Mark stars with the FAINT_STAR_COLOR if they have MAG_ACA
+        # that is fainter than red_mag_lim but MAG_ACA - a rough mag error
+        # is brighter than that limit.  The rough mag error calculation is
+        # based on the SAUSAGE acq stage 1 check, which uses nsigma of
+        # 3.0, a mag low limit of 1.5, and a random error of 0.26.
+        nsigma = 3.0
+        mag_error_low_limit = 1.5
+        randerr = 0.26
+        caterr = stars['MAG_ACA_ERR'] / 100.
+        error = nsigma * np.sqrt(randerr**2 + caterr**2)
+        error = np.clip(error, a_min=mag_error_low_limit, a_max=None)
+        faint = ((stars['MAG_ACA'] >= red_mag_lim)
+                 & ((stars['MAG_ACA'] - error) < red_mag_lim))
+        # Faint and bad stars will keep their BAD_STAR_COLOR
+        # Only use the faint mask on stars that are not bad
+        faint = faint & ~bad_stars
```

Comment:
You can remove this line and the comment by moving the `colors[bad_stars] = BAD_STAR_COLOR` statement after this block.